<launch>
  <!-- Load URDF for simulation -->
  <param name="robot_description" command="$(find xacro)/xacro '$(find husky_description)/urdf/husky.urdf.xacro'" />

  <!-- Start Gazebo with Husky in playpen world -->
  <include file="$(find husky_gazebo)/launch/husky_playpen.launch" />
  
  <!-- Start move_base with navigation configuration -->
  <!--<include file="$(find husky_navigation)/launch/move_base.launch" /> -->

  <!-- Static transform: map -> odom (Option 2 Fix) 
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom_broadcaster"
        args="0 0 0 0 0 0 map odom 10" />-->

<!-- Visualize saved waypoints -->
  <node pkg="husky_waypoint_nav" type="waypoint_visualizer.py" name="waypoint_visualizer" output="screen">
    <param name="/outdoor_waypoint_nav/coordinates_file" value="$(find husky_waypoint_nav)/config/optitrack_waypoints.txt"/>
  </node>


  <!-- RViz (optional) -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find husky_viz)/rviz/navigation.rviz"
        required="false" output="screen" />

  <!-- Simulated OptiTrack Pose Bridge -->
  <node pkg="husky_waypoint_nav" type="gazebo_to_pose_bridge" name="gazebo_to_pose_bridge" output="screen" />

  <!-- Dummy Waypoint Generator for Simulation -->
  <node pkg="husky_waypoint_nav" type="generate_dummy_waypoints.py" name="generate_dummy_waypoints" output="screen" />

  <!-- Waypoint Navigation Node -->
  <node pkg="husky_waypoint_nav" type="optitrack_waypoint_navigator" name="optitrack_waypoint_navigator" output="screen">
    <param name="outdoor_waypoint_nav/coordinates_file" value="$(find husky_waypoint_nav)/config/optitrack_waypoints.txt"/>
    <param name="outdoor_waypoint_nav/loop_mode" value="true"/>
    <param name="outdoor_waypoint_nav/pause_button_num" value="3"/>
    <param name="outdoor_waypoint_nav/resume_button_num" value="1"/>
  </node>
</launch>

